packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -3006,6 +3064,42 @@ function beginWork(
                 renderExpirationTime,
               );
             } else {
+              // The primary child fragment does not have pending work marked
+              // on it...
+
+              // ...usually. There's an unfortunate edge case where the fragment
+              // fiber is not part of the return path of the children, so when
+              // an update happens, the fragment doesn't get marked during
+              // setState. This is something we should consider addressing when
+              // we refactor the Fiber data structure. (There's a test with more
+              // details; to find it, comment out the following block and see
+              // which one fails.)
+              //
+              // As a workaround, we need to recompute the `childExpirationTime`
+              // by bubbling it up from the next level of children. This is
+              // based on similar logic in `resetChildExpirationTime`.
```

Comment:
An alternative fix would be to detect this during `setState`. I thought it might be better to localize the fix here since it's relatively rare but I'm having second thoughts.

Reply:
I think this is the right call but I think we should just bite the bullet have have Suspense boundaries always have these fragments even when not suspended. It's too much code to manage (buggy and byte size). Especially with enter/exit animations it's going to get even trickier.

It adds a bit more memory but there's not that many suspense boundaries and the ones that do exist fairly often transition through this state anyway.

Reply:
That would definitely simplify things a lot